### DISCUSSION TITLE
Using over VPN

### DISCUSSION THREAD

**Discussion by y20zhang-nokia** [debauchee/barrier#1282] [Q&A]
### Operating Systems ###

Server/Client: Ubuntu 18.04 or Manjaro 20.01 or Windows 10

Is there any suggestion for using barrier over VPN? I lost control immediately after turning the VPN on.
I saw this [post](https://mrfrosti.com/2017/07/31/using-synergy-over-vpn/) about Synergy over VPN, but I am not sure how applicable it is.

**Comment by p12tic** [comment]
You need to still have routing to LAN available. I suspect your VPN solution redirects all network traffic through the VPN.

Barrier works through a VPN that only affects certain network subnets just fine, I'm using it regularly this way.

**Reply by rpg-hub** [reply]
But how do you make your Keyboard and mouse work?
My KVM server (Barrier server) is Win 10. laptop is Barrier client (win 11).
It works fine but when i connect to VPN on laptop I can not use Keyboard and Mouse from my Win10 PC anymore....

**Reply by fkieber** [reply]
Have you try my solution below with ssh or PuTTY ?
You have to launch this on your laptop client.

**Reply by xNS5** [reply]
That's probably because your VPN application is running full-tunnel, meaning your computer's internet traffic will get routed 100% through the host server. If you have the option to, run it in split-tunnel mode. At least, that was the case in my personal experience.

**Comment by fkieber** [comment]
I use successfully an ssh tunnel.
ssh -g -C -T -p 22 -L 24800:my_barrier_server_local_network_ip:24800  user@my_server_external_ip
On Windows you can use PuTTY.

**Reply by raspbounty** [reply]
I am trying to get your solution working but I use PuTTY and do not really know what options I need to use. Could you tell me what options I have to enter in which option?

**Reply by fkieber** [reply]
Here is a screenshot that show how you can do in Putty. Fill in the fields, than click on "Add"
![2022-04-27_11h12_07](https://user-images.githubusercontent.com/493622/165485684-0e3429c0-e6b7-4fc5-b8f4-9c89344e6a85.png)

An other solution is to use ssh in Cygwin. A little more heavy but you can use command line like in Linux.
